packages/components/addon/components/hds/tabs/index.hbs

```diff
@@ -0,0 +1,25 @@
+{{! template-lint-disable no-invalid-role }}
+{{! template-lint-disable require-context-role }}
+
+<div class={{this.classNames}} ...attributes>
+  {{! Hds-TabList }}
+  <div class="hds-tab-list hds-typography-body-200">
+    <ul class="hds-tab-list__list" role="tablist">
+      {{yield
+        (hash Tab=(component "hds/tabs/tab" didInsertNode=this.didInsertTab panelIds=this.panelIds tabIds=this.tabIds))
+      }}
+    </ul>
+  </div>
+
+  {{yield
+    (hash
+      Panel=(component
+        "hds/tabs/panel"
+        didInsertNode=this.didInsertPanel
+        tabIds=this.tabIds
+        panelIds=this.panelIds
+        panelNodes=this.panelNodes
+      )
+    )
+  }}
```

Comment:
Right now "Tab" and "Panel" are separate components. I was also considering a different approach where users could add "Content" (or whatever) components which would set both Tab titles and accept content similar to Panel components to simplify usage for end-users.

Something like:
```
<Tabs>
  <TabContent @title="Tab 1 label">
    Add any text etc here which appears in "Panel"
  </TabContent>

  <TabContent @title="Tab 2 label">
    Add any text etc here which appears in "Panel"
  </TabContent>
</Tabs>
```

Reply:
I think is a very clever idea, and I am tempted to test/pursue it (this approach would allow us to automatically associate the tab with the corresponding content). I see though some possible downsides, that we should consider:
- [major] I am not sure we would have access in the controller/JS file to the children of `Tabs`; so the JS code may become convoluted when we need to switch active tabs (but maybe not; probably we have to test) - @alex-ju what to you think?
- [intermediate] users may want to use a wrapping element around the content/panels, for some reasons (eg, custom styling, positioning, special layout needs, many other things) and maybe this approach would not allow it (or maybe yes because there's a `yield`? we have to test how it works in Ember) - _btw, this is not a problem of this specific approach, but something we have to consider for all the different approaches_
- [minor] having the content of the tab `@title` will prevent us from having structured content (eg HTML or components) inside the `tab-text` instead of just plain text; I know it _should_ be only text, it already happened with the `label` in the Form control that we discovered later we needed to support also nodes (luckily for us, our implementation supported it) - _Ember is not like React, you can't pass HTML/Components inside a prop 🤷‍♂️_

Reply:
Regarding point n.2 above, I have made a quick test with the current approach you're using in your PR, and is possible to wrap the `<T.Panel>` elements in a wrapper, and it **kinda** works (not perfect though, but could be an acceptable compromise)
That said, looking at _how_ it works (very bizarre!) I suspect the alternative approach you suggested above would not allow a wrapper only for the content, because it would wrap both the tabs and content in the same container (there would be only one `yield` for both, inside the `Tabs`.

When I say bizarre, try for yourself: use this code:
```
<Hds::Tabs as |T|>
  <div class="aaa">
    <T.Tab>One</T.Tab>
    <T.Tab>Two</T.Tab>
    <T.Tab>Three</T.Tab>
  </div>
  <div class="bbb">
    <T.Panel>Content one</T.Panel>
    <T.Panel>Content 2</T.Panel>
    <T.Panel>Content 3</T.Panel>
  </div>
</Hds::Tabs>

```
and see how the content is rendered, and where!

Spoiler:
<img width="694" alt="modified" src="https://user-images.githubusercontent.com/686239/189113287-e96d1a8a-559e-4318-bc57-bd20859fc437.png">

Reply:
Thanks. I'll take a closer look. The Tabs need to be able to optionally accept icons and or counter badges though so that could be more difficult with the alternate structure I proposed.

Reply:
@didoo I also looked at the Tabs component in Consul you recommended I look at: https://github.com/hashicorp/consul/tree/2b8273bc6e45cc0ea8b84976a02c2a92c6aa1071/ui/packages/consul-ui/app/components/tab-nav

I don't know if that would be a better model than what I've been doing so far? It does seem simpler.

It would be great to hear feedback on approaches from @MelSumner and @alex-ju too if either of you have some time to take a look. Thanks!

Reply:
> The Tabs need to be able to optionally accept icons and or counter badges though so that could be more difficult with the alternate structure I proposed.

Not necessarily, if the only things that are needed are the icon name and the count "text/value" then they could be props exposed, and passed down to the `<FlightIcon>` and `<Hds::BadgeCount>` subcomponents

> I also looked at the Tabs component in Consul […] I don't know if that would be a better model than what I've been doing so far? It does seem simpler.

My recommendation was about the `indicator` implementation and animation that I found interesting. The rest of the code I think is too distant from our approach to components, to be applicable to us

Reply:
Ok thanks for that clarification

Reply:
The way I envisaged the API for a tabs component in this library would be like this:
```
<Hds::Tabs as |T|>
  <T.Tab @label="Label 1" @isSelected={{true}}>
    Sample tab content 1
  </T.Tab>
  <T.Tab @label="Label 2">
    Sample tab content 2
  </T.Tab>
</Hds::Tabs>
```
Which is very close to [your suggestion above](https://github.com/hashicorp/design-system/pull/566#discussion_r965328099), Kristin.

The advantages I see with it are that we avoid any potential mismatches between labels/tabs and panels and we have full control on how to create relationship between them. It also has the side-effect of constrain the label to text, which invites the question of how to achieve a badge in a tab label.

I'm thinking we can use the component helper to allow a `<BadgeCount>` or an icon before or after the label, as shown below.
```
<Hds::Tabs as |T|>
  <T.Tab @label="Label 1" @labelSuffix={{component 'badge-count' text="3" size="small"}} @isSelected={{true}}>
    Sample tab content 1
  </T.Tab>
  <T.Tab @label="Label 2">
    Sample tab content 2
  </T.Tab>
</Hds::Tabs>
```

If the above suggestion strays to far from your vision, an alternative suggestion would be
```
<Hds::Tabs as |T|>
  <T.Tab @isSelected={{true}}>
    Sample tab content 1
  </T.Tab>
  <T.Tab>
    Sample tab content 2
  </T.Tab>
  <T.Panel>
    Sample panel content 1
  </T.Panel>
  <T.Panel>
    Sample panel content 2
  </T.Panel>
</Hds::Tabs>
```

which would require some convention to create relationship between tabs and panels (either using the order, or an attribute), something that you've already considered in this PR.

Reply:
@alex-ju looking at the Figma designs, my understanding is that the icon is always leading (with color inherited from the text) and the badge is always trailing (with fixed size and color). maybe @heatherlarsen can confirm this?

if this is the case, the API can be simplified even more:
```
<Hds::Tabs as |T|>
  <T.Tab @label="Label 1" @count="3" @icon="arrow-left" @isSelected={{true}}>
    Sample tab content 1
  </T.Tab>
  ...
</Hds::Tabs>
```

Reply:
@alex-ju @KristinLBradley another consideration re. the pattern/API above, in this case about the `label`.
I am _really_ worried that we will find ourselves in need to add "stuff" to the label that is not just text, and in that case I could imagine we could use a contextual component as alternative (in the same way we did for the `Error/ErrorMessages` in the form controls. Something like this:
```
<Hds::Tabs as |T|>
  <T.Tab @count="3" @icon="arrow-left" @isSelected={{true}} as |TT|>
    <TT.Label>This is some text <SomeComponentOrHTMLElement></TT.Label>
    Sample tab content 1
  </T.Tab>
  ...
</Hds::Tabs>
```
but in this case I _think_ the different `yield`s will interfere one with another, and it would not work. So I was thinking if we should instead wrap the content in a specific contextual component, so that both these results would (I think) work:
```
<Hds::Tabs as |T|>
  <T.Tab @label="my label" @count="3" @icon="arrow-left" @isSelected={{true}} as |TT|>
    <TT.PanelContent>Sample tab content 1</TT.PanelContent>
  </T.Tab>
  ...
</Hds::Tabs>
```
and
```
<Hds::Tabs as |T|>
  <T.Tab @count="3" @icon="arrow-left" @isSelected={{true}} as |TT|>
    <TT.Label>This is some text <SomeComponentOrHTMLElement></TT.Label>
    <TT.PanelContent>Sample tab content 1</TT.PanelContent>
  </T.Tab>
  ...
</Hds::Tabs>
```

Reply:
@alex-ju @didoo Thanks for the additional input. I think I will continue with the current implementation I started on to begin and then if there's time I may try to refactor following my 2nd idea or a variation of that as @didoo proposed.